controllers/kustomization_impersonation.go

```diff
@@ -177,15 +177,15 @@ func (ki *KustomizeImpersonation) getKubeConfig(ctx context.Context) ([]byte, er
 	}
 
 	var kubeConfig []byte
-	for k := range secret.Data {
-		if k == "value" || k == "value.yaml" {
-			kubeConfig = secret.Data[k]
-			break
-		}
-	}
-
-	if len(kubeConfig) == 0 {
-		return nil, fmt.Errorf("KubeConfig secret '%s' doesn't contain a 'value' key ", secretName.String())
+	if key := ki.kustomization.Spec.KubeConfig.SecretRef.Key; key != "" {
```

Comment:
Let's rewrite this as a `switch` please.

Reply:
I don't know if that is possible since we don't have a single value to switch on.

PS: This part was tricky to preserve existing behavior. We used to loop through `secret.Data` looking for a key "value" or "value.yaml" which meant that if both were present, the the one picked would indeterministic since ranging over a map is not an ordered operation. I removed the loop, and tried to implement deterministic logic as: prefer user-specified key, fallback to "value" key, fallback to "value.yaml" key.

Open to code suggestions.

Reply:
One option would be:

```go
switch {
case ki.kustomization.Spec.KubeConfig.SecretRef.Key != "":
	kubeConfig = ki.kustomization.Spec.KubeConfig.SecretRef.Key
case secret.Data["value"] != "":
	kubeConfig = secret.Data["value"]
case secret.Data["value.yaml"] != "":
	kubeConfig = secret.Data["value.yaml"]
default:
	// User did not specify a key, and the 'value' key was not defined.
	return nil, fmt.Errorf("KubeConfig secret '%s' does not contain a 'value' key", secretName)
}
```

but not sure if this is actually better, although it would cover for keys with an empty value.

Reply:
Updated to use switch